src/Microsoft.Health.Fhir.Liquid.Converter/Cda/CdaDataParser.cs

```diff
@@ -0,0 +1,121 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Cda
+{
+    public class CdaDataParser
+    {
+        public IDictionary<string, object> Parse(string document)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(document))
+                {
+                    throw new DataParseException(FhirConverterErrorCode.NullOrEmptyInput, Resources.NullOrEmptyInput);
+                }
+
+                var xDocument = XDocument.Parse(document);
+
+                // Remove redundant namespaces to avoid appending namespace prefix before elements
+                var defaultNamespace = xDocument.Root?.GetDefaultNamespace().NamespaceName;
+                xDocument.Root?.Attributes()
+                    .Where(attribute => IsRedundantNamespaceAttribute(attribute, defaultNamespace))
+                    .ToList()
+                    .ForEach(action: x => x.Remove());
```

Comment:
Is it ok for a XDocument to remove nodes while iterating  and can we add relate test case for such scenarios?

Reply:
Yes, it is OK to do so, but after checking the interfaces I changed it into `IEnumerable<XAttribute> Remove`, which might be better. More tests are also added for the parser.